packages/babel-plugin-transform-react-jsx-source/src/index.js

```diff
@@ -28,6 +28,21 @@ export default function ({ types: t }) {
 
   let visitor = {
     JSXOpeningElement(path, state) {
+      const id = t.jSXIdentifier(TRACE_ID);
+      const location = path.container.openingElement.loc;
+      if (!location) {
+        // the element was generated and doesn't have location information
+        return;
+      }
+
+      const attributes = path.container.openingElement.attributes;
+      for (let i = 0; i < attributes.length; i++) {
+        if (attributes[i].name.name === TRACE_ID) {
```

Comment:
I think we need to check for `attributes[i].name` or filter out JSXSpreadAttribute?